product_listing.go

```diff
@@ -0,0 +1,147 @@
+package goshopify
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const productListingBasePath = "product_listings"
+const productsListingResourceName = "product_listings"
+
+// linkRegex is used to extract pagination links from product search results.
+//var linkRegex = regexp.MustCompile(`^ *<([^>]+)>; rel="(previous|next)" *$`)
+
+// ProductListingService is an interface for interfacing with the product listing endpoints
+// of the Shopify API.
+// See: https://help.shopify.com/api/reference/product_listings
```

Comment:
That url returns a 404, perhaps replace with this reference:
`https://shopify.dev/docs/admin-api/rest/reference/sales-channels/productlisting`